database/migrations/2021_07_21_085048_create_colors_table.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateColorsTable extends Migration
+{
+	/**
+	 * Run the migrations.
+	 *
+	 * @return void
+	 */
+	public function up()
+	{
+		Schema::create('colors', function (Blueprint $table) {
+			$table->bigIncrements('id');
+			$table->unsignedTinyInteger('r')->nullable();
+			$table->unsignedTinyInteger('g')->nullable();
+			$table->unsignedTinyInteger('b')->nullable();
+			$table->foreignId('photo_id')->constrained();
```

Comment:
Add `->onUpdateCascade()` and `->onDeleteCascade()` here. Then you can drop the `boot` method and the listener for the deletion event from the model `Photo`.